.github/workflows/continous-integration-docker.yml

```diff
@@ -491,7 +449,94 @@ jobs:
       height_grep_text: 'current_height.*=.*Height.*\('
     secrets: inherit
 
-  # TODO: testnet checkpoints, test-full-sync, test-update-sync, get-available-disks (+ reusable)
+  # zebra testnet checkpoint generation tests
+  #
+  # These tests will fail when testnet is unstable, they should not be required to merge.
+  #
+  # TODO: ignore failures on testnet, so any failures don't appear in the GitHub interface.
+
+  # Test that Zebra can run a full testnet sync, and regenerate chain tip Zebra cached state disks.
+  # This job always runs on testnet, regardless of any inputs or variable settings.
+  #
+  # Runs:
+  # - after every PR is merged to `main`
+  # - on every PR update
+  # - in manual workflow runs, but only if the state version in constants.rs has no cached disk
+  # TODO: move to the main branch if it's too slow:
+  # - on every PR update, but only if the state version in constants.rs has no cached disk
+  #
+  # Note: the output from get-available-disks-testnet should match with the caller workflow inputs
+  test-full-sync-testnet:
+    name: Zebra tip on testnet
+    needs: [ build, get-available-disks-testnet ]
+    uses: ./.github/workflows/deploy-gcp-tests.yml
+    # to also run on Mergify head branches,
+    # add `|| (github.event_name == 'push' && startsWith(github.head_ref, 'mergify/merge-queue/'))`:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-based-on-the-head-or-base-branch-of-a-pull-request-1
+    #
+    # to only run on the main branch:
+    #if: ${{ (github.event_name == 'push' && github.ref_name == 'main') || !fromJSON(needs.get-available-disks-testnet.outputs.zebra_tip_disk) || github.event.inputs.run-full-sync == 'true' }}
+    # Runs on every PR, and in manual runs if the disks are missing
+    if: ${{ (github.event_name == 'push' || !fromJSON(needs.get-available-disks-testnet.outputs.zebra_tip_disk)) && github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true' && github.event.inputs.run-lwd-sync != 'true' }}
+    with:
+      app_name: zebrad
+      test_id: full-sync-to-tip-testnet
+      test_description: Test a full sync up to the tip on testnet
+      # The value of FULL_SYNC_TESTNET_TIMEOUT_MINUTES is currently ignored.
+      test_variables: '-e TEST_FULL_SYNC=1 -e ZEBRA_FORCE_USE_COLOR=1 -e FULL_SYNC_TESTNET_TIMEOUT_MINUTES=0'
+      network: 'Testnet'
+      # This test should be quick, only an hour or so
+      is_long_test: false
+      needs_zebra_state: false
+      saves_to_disk: true
```

Comment:
TODO before merging this PR:
- Periodically delete the testnet disks

This will happen on my next push to this PR, but I want to see if the sync works first.

Reply:
This is on #6582 because it's actually going to happen anyway, but slightly wrong, due to the way the delete job is written.